specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2023-12-10-preview/cloudhsm.json

```diff
@@ -0,0 +1,967 @@
+      "get": {
+        "tags": [
+          ""
+        ],
+        "operationId": "PrivateEndpointConnections_ListByCloudHsmCluster",
+        "description": "The List operation gets information about the private endpoint connections associated with the Cloud HSM Cluster",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/cloudHsmClusterName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "List of all CloudHsmCluster private endpoint connections associated with the Cloud HSM Cluster.",
+            "schema": {
+              "$ref": "#/definitions/PrivateEndpointConnectionListResult"
+            }
+          },
+          "default": {
+            "description": "The error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudHsmClusterError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "CloudHsmClusterPrivateEndpointConnection_ListByResource_MaximumSet_Gen": {
+            "$ref": "./examples/CloudHsmClusterPrivateEndpointConnection_ListByCloudHsmCluster_MaximumSet_Gen.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CloudHsmProperties": {
+      "type": "object",
+      "description": "The Cloud HSM Properties",
+      "properties": {
+        "fqdn": {
+          "type": "string",
+          "description": "FQDN of the Cloud HSM"
+        },
+        "state": {
+          "type": "string",
+          "description": "The Cloud HSM State"
+        },
+        "stateMessage": {
+          "type": "string",
+          "description": "The Cloud HSM State message"
+        }
+      }
+    },
+    "CloudHsmCluster": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/CloudHsmClusterProperties",
+          "description": "Properties of the Cloud HSM Cluster"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/CloudHsmClusterResource"
+        }
+      ],
+      "description": "Resource information with extended details.",
+      "type": "object"
+    },
+    "CloudHsmClusterResource": {
+      "properties": {
+        "sku": {
+          "$ref": "#/definitions/CloudHsmClusterSku",
+          "description": "SKU details"
+        },
+        "identity": {
```

Comment:
Can I assume, this is optional ?

Reply:
yes, the required properties have an additional tag "required"